app/controllers/groups_controller.rb

```diff
@@ -145,10 +150,16 @@ def delete_vpn
   end
 
   def add_group
-    @user = User.find(params[:id])
+    user_id = params[:id]
     if current_user.admin?
-      @group = Group.find(params[:group_id])
-      @group.add_user(@user.id)
+      begin
+        expiration_date = expiration_date_param
+      rescue ArgumentError
```

Comment:
Haven't found any way to remove `exception as control flow` because date validation is done by parsing the param.
Need some advice to work around it.

Reply:
oh yeah.. this is date.. 
then it's cannot be helped..